### PR TITLE
PWA-464 pdp implement the basic route

### PR DESCRIPTION
**Pull request by devbucket**

### CODE REVIEW

File: pages/Product/components/Content/connector.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+});
+
+/**
+ * Check to see if the product name arrived.
+ * @param {*} next The next props.
+ * @param {*} prev The previous props.
+ * @returns {boolean}
+ */
+const areStatePropsEqual = (next, prev) => {
+  if (!prev.name && next.name) return false;
```
Comment (richardgorman) [inline pages/Product/components/Content/connector.js]: Please expand all new conditions to use multiple lines.

File: pages/Product/components/Content/connector.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * @param {Object} state The current application state.
+ * @return {Object} The extended component props.
+ */
+const mapStateToProps = state => ({
+  name: getProductName(state),
+});
+
+/**
+ * Check to see if the product name arrived.
+ * @param {*} next The next props.
```
Comment (richardgorman) [inline pages/Product/components/Content/connector.js]: Type is not *

File: components/ProductGrid/components/Item/index.jsx [hunk 3]
```diff
@@ -16,110 +17,117 @@ import FavoritesButton from '@shopgate/pwa-ui-shared/FavoritesButton';
 import styles from './style';
 import connect from './connector';
 
+const props = {};
```
Comment (richardgorman) [inline components/ProductGrid/components/Item/index.jsx]: Why is this inside this ticket? We have no good way to test this out yet.

File: pages/Product/components/Content/connector.js (hunk 1 above)
Comment (richardgorman) [inline pages/Product/components/Content/connector.js]: What is name needed for? If it's just the test id then remove it and QA needs to find another way.

File: pages/Product/streams.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { routeDidEnter } from '@shopgate/pwa-common/streams/history';
+import { routeWillEnter$ } from '@shopgate/pwa-common/streams/router';
+import { ITEM_PATH } from '@shopgate/pwa-common-commerce/product/constants';
+
+export const productWillEnter$ = routeWillEnter$
+  .filter(({ action }) => (
+    action.route.pathname.startsWith(ITEM_PATH) &&
```
Comment (richardgorman) [inline pages/Product/streams.js]: Please change the ITEM_PATH constant to include the :productId param. Then you can just test the pattern against it.

File: pages/Product/subscriptions.js [hunk 5]
```diff
@@ -1,85 +1,76 @@
...
-import { routeDidEnter } from '@shopgate/pwa-common/streams/history';
-import appConfig from '@shopgate/pwa-common/helpers/config';
-import { getCurrentProductVariantId } from '@shopgate/pwa-common-commerce/product/selectors/variants';
-import { getCurrentBaseProductId } from '@shopgate/pwa-common-commerce/product/selectors/product';
-import { successReviewSubmit$ } from '@shopgate/pwa-common-commerce/reviews/streams';
+// Import { main$ } from '@shopgate/pwa-common/streams/main';
+// Import { routeDidEnter } from '@shopgate/pwa-common/streams/history';
+// Import appConfig from '@shopgate/pwa-common/helpers/config';
+import { hex2bin } from '@shopgate/pwa-common/helpers/data';
+import setViewTitle from '@shopgate/pwa-common/action-creators/view/setViewTitle';
```
Comment (richardgorman) [inline pages/Product/subscriptions.js]: Please use setTitle action instead.

File: pages/Product/components/Content/connector.js (hunk 1 above)
Comment (devbucket) [reply pages/Product/components/Content/connector.js]: No. I won't remove this now. We can't open a new construction site for QA now. This is the only thing he found in long time.

Comment (devbucket) [reply pages/Product/components/Content/connector.js]: Done

File: components/ProductGrid/components/Item/index.jsx (hunk 3 above)
Comment (devbucket) [reply components/ProductGrid/components/Item/index.jsx]: Done

File: pages/Product/streams.js (hunk 4 above)
Comment (devbucket) [reply pages/Product/streams.js]: Done

File: pages/Product/subscriptions.js (hunk 5 above)
Comment (devbucket) [reply pages/Product/subscriptions.js]: Done

**Pull request merged**